### ISSUE TITLE
`popup_centered()` not centered (usually offscreen)

### ISSUE THREAD

**Issue by benbusby** [godotengine/godot#67002]
### Godot version

v4.0.beta2.official [f8745f2f7]

### System information

macOS 12.6

### Issue description

Using `popup_centered()` to open a FileDialog does not center the dialog, but instead always places it in the same place nearly offscreen. This also applies to popups within the FileDialog itself, such as the file type menu.

I also see the following error:

```
E 0:00:01:0620   _on_button_pressed: Window 0 spawned at invalid position: (764, 459).
  <C++ Source>   scene/main/window.cpp:1270 @ popup()
  <Stack Trace>  demo-button.gd:6 @ _on_button_pressed()
```

Screenshots:

![Screen Shot 2022-10-06 at 3 05 39 PM](https://user-images.githubusercontent.com/33362396/194419935-3913ff2d-35d9-4841-b6cf-9e187b00e637.png)
![Screen Shot 2022-10-06 at 3 12 09 PM](https://user-images.githubusercontent.com/33362396/194419940-1c263b33-eb47-4938-bde0-7b609cb28641.png)

### Steps to reproduce

If using the attached reproduction project:

1. Open the attached project and run the scene
2. Click the button to open the file dialog
3. Notice that it isn't centered, and dialogs within the FileDialog are also placed in random areas.

If re-creating from scratch:

1. Create a scene with a Button and FileDialog
2. Attach a signal to open the FileDialog using the button's `pressed` signal
3. Follow the same steps for attached project

### Minimal reproduction project

[filebrowserbug.zip](https://github.com/godotengine/godot/files/9728689/filebrowserbug.zip)

**Comment by harrisyu** [comment]
I have same problem in project manager, maybe it's cause by multi-monitor setting in windows.

**Comment by slmnll** [comment]
I have same problem on my mac as well. It occurs when the game window is placed in a sub-monitor in my case.

**Comment by Cykyrios** [comment]
This should be fixed by #66712, which will be in beta3.

**Comment by benbusby** [comment]
It does appear that #66712 fixed the issue -- I'm no longer seeing the bug in beta3.
